### PR TITLE
Add topic detection feature to batch transcribe

### PR DESCRIPTION
**Pull request by aaronng91**

### CODE REVIEW

File: speechmatics/cli_parser.py [hunk 1]
```diff
@@ -283,6 +283,24 @@ def get_arg_parser():
         help="Perform sentiment analysis on the transcript",
     )
 
+    # Parent parser for batch topic-detection argument
+    batch_topic_detection_parser = argparse.ArgumentParser(add_help=False)
+    batch_topic_detection_parser.add_argument(
```
Comment (TudorCRL) [inline speechmatics/cli_parser.py:288]: I don't know much about the API, but I'm guessing it works with sensible defaults even if you don't provide a custom list of topics?

File: speechmatics/cli_parser.py [hunk 2]
```diff
@@ -283,6 +283,24 @@ def get_arg_parser():
...
+        "--topic-detection",
+        dest="topic_detection",
+        action="store_true",
+        default=False,
+        help="Whether to detect topics in transcript",
+    )
+    batch_topic_detection_parser.add_argument(
+        "--topic-detection-topics",
+        dest="topics",
```
Comment (TudorCRL) [inline speechmatics/cli_parser.py:297]: If the destination var is topics, does the flag need to be --topic-detection-topics or could it just be topics? Is there any possibility we'll have topics as an input to some other thing?

File: speechmatics/cli_parser.py (hunk 1 above)
Comment (HennerM) [inline speechmatics/cli_parser.py:288]: for a CLI flag `speechmatics transcribe --detect-topics` reads nicer than `--topic-detection`. It helps the user know that it's something on the transcription IMO

File: speechmatics/cli_parser.py (hunk 2 above)
Comment (HennerM) [reply speechmatics/cli_parser.py:297]: argparse has this nice feature that you can also use abbreviations if it isn't ambigiuous, e.g. `--topic science` would work for `--topic-detection-topics`, but yes I agree that the name is a bit weird.
What about `--topics-to-detect`?

Comment (dumitrugutu) [reply speechmatics/cli_parser.py:297]: We can use `--enable-topic-detection` to enable the feature with default topics or `--enable-topic-detection --topics science` as a combination.

File: speechmatics/cli_parser.py (hunk 1 above)
Comment (aaronng91) [reply speechmatics/cli_parser.py:288]: yes, we do have a default list of 10 topics baked into the feature, which will be used when user does not pass in a list of topics

File: speechmatics/cli.py [hunk 3]
```diff
@@ -325,6 +326,18 @@ def get_transcription_config(
...
         config["sentiment_analysis_config"] = SentimentAnalysisConfig()
 
+    topic_detection_config = config.get("topic_detection_config", {})
+    args_topic_detection = args.get("topic_detection")
+    if args_topic_detection or topic_detection_config:
+        topic_detection_config = TopicDetectionConfig()
+        topics = (
+            args_topic_detection.split(",")
+            if args_topic_detection
+            else topic_detection_config.get("target_languages")
```
[the remark this answers is not in the document]
Comment (aaronng91) [reply speechmatics/cli.py]: thanks for spotting this!

File: speechmatics/cli.py [hunk 4]
```diff
@@ -325,6 +326,18 @@ def get_transcription_config(
     if args_sentiment_analysis or sentiment_analysis_config:
         config["sentiment_analysis_config"] = SentimentAnalysisConfig()
 
+    topic_detection_config = config.get("topic_detection_config", {})
+    args_topic_detection = args.get("topic_detection")
+    if args_topic_detection or topic_detection_config:
+        topic_detection_config = TopicDetectionConfig()
+        topics = (
+            args_topic_detection.split(",")
```
[the remark this answers is not in the document]
Comment (aaronng91) [reply speechmatics/cli.py]: good spot, fixed

File: speechmatics/cli_parser.py (hunk 1 above)
Comment (aaronng91) [reply speechmatics/cli_parser.py:288]: agree. it's just that --sentiment-analysis seems a bit out of place now, I wish it was standardized. But at least we follow the --summarize flag

File: speechmatics/cli_parser.py (hunk 2 above)
Comment (aaronng91) [reply speechmatics/cli_parser.py]: a couple of different ideas here, can we settle for --detect-topics and --topics XX YY ZZ?

Comment (TudorCRL) [reply speechmatics/cli_parser.py]: I'm happy with that

**Comment by aaronng91** [comment]
may I get approval? @TudorCRL @dumitrugutu @HennerM

**Review by HennerM — APPROVED** [review]

**Pull request merged**